package/MDAnalysis/core/groups.py

```diff
@@ -1641,12 +1644,45 @@ def unwrap(self, compound='fragments', reference='com', inplace=True):
                     refpos = refpos.astype(np.float32, copy=False)
                     target = distances.apply_PBC(refpos, self.dimensions)
                     positions[mask] += target - refpos
+
+
         if inplace:
             unique_atoms.positions = positions
         if not atoms.isunique:
             positions = positions[atoms._unique_restore_mask]
         return positions
 
+    def choose_periodic_image(self, reference_point, centre='com', inplace=True):
+        atoms = self.atoms
+
+        unique_atoms = atoms.unique
+
+        if centre is not None:
+            ref = centre.lower()
+            if ref  == 'com':
+                # Don't use hasattr(self, 'masses') because that's incredibly
+                # slow for ResidueGroups or SegmentGroups
+                if not hasattr(unique_atoms, 'masses'):
+                    raise NoDataError("Cannot perform choose_image with "
+                                      "reference='com', this requires masses.")
+                ctrpos = unique_atoms.center_of_mass(pbc=False)
+            elif ref == 'cog':
+                ctrpos = unique_atoms.center_of_geometry(pbc=False)
+            else:
+                raise ValueError("Unrecognized reference '{}'. Please use one "
+                                 "of 'com', 'cog', or None.".format(centre))
+
+        difference = reference_point - ctrpos
+
+        shift = np.rint(np.divide(difference, unique_atoms.dimensions[:3]))
```

Comment:
@NinadBhat this only works for orthogonal boxes, for triclinic boxes the math is a little more complicated:

https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/lib/include/calc_distances.h#L34